### PR TITLE
Added search similar people by Name, Gender and BirthYear

### PR DESCRIPTION
**Pull request by corneliouzbett**
@ibacher We should standard [OpenmrsCodeFormatter](https://github.com/openmrs/openmrs-core/blob/master/tools/src/main/resources/eclipse/OpenMRSFormatter.xml)

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPatientDaoImpl.java [hunk 1]
```diff
@@ -80,13 +80,13 @@ public PatientIdentifierType getPatientIdentifierTypeByNameOrUuid(String name, S
 	@SuppressWarnings("unchecked")
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPatientDaoImpl.java]: Can you remove this file from this commit? It doesn't seem like you changed anything relevant here.

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PersonTranslatorImpl.java [hunk 2]
```diff
@@ -31,16 +31,16 @@
 @Component
 @Setter(AccessLevel.PACKAGE)
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PersonTranslatorImpl.java]: Can you remove this file from this PR? It doesn't seem that anything changed.

File: api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPersonServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	private static final String PERSON_NAME = "John";
+	
+	private static final String PERSON_GENDER = "M";
+	
+	private static final int PERSON_BIRTH_YEAR = 1999;
+	
+	private static final String WRONG_PERSON_UUID = "wrong_person_uuid";
+	
+	@Inject
+	private FhirPersonService personService;
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPersonServiceImplTest.java]: Ideally, outside of the tests for the Daos, we shouldn't be relying on the actual database for anything. That means that to test the service, we should pass in a mock FhirPersonDao that returns the expected results, similar to what's done for PatientTranslatorImplTest.

File: api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/PersonTranslatorImplTest.java [hunk 4]
```diff
@@ -37,45 +37,45 @@
 
 @RunWith(MockitoJUnitRunner.class)
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/PersonTranslatorImplTest.java]: Can you remove this file from this PR? It doesn't seem that anything changed.

File: omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java [hunk 5]
```diff
@@ -16,14 +16,14 @@
 import org.openmrs.api.context.Context;
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java]: Can you remove this file from this PR? It doesn't seem that anything changed.

File: omod/src/main/java/org/openmrs/module/fhir2/providers/PersonFhirResourceProvider.java [hunk 6]
```diff
@@ -28,21 +34,42 @@
...
+	 * @param name Name of the person to search
+	 * @param birthDate The year of birth
+	 * @param gender Gender field to search on (Typically just "M" or "F")
+	 * @return Returns a bundle list of people. This list may contain multiple matching
+	 * 	 * resources, or it may also be empty.
+	 */
+
+	@Search
+	@SuppressWarnings("unused")
+	public Bundle findSimilarPeople(@RequiredParam(name = Person.SP_NAME) StringParam name,
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/fhir2/providers/PersonFhirResourceProvider.java]: Do these actually need to be required parameters or should they be optional parameters?

**Comment by corneliouzbett** [comment]
I found this useful since it's implemented in openmrs Service layer. Furthermore, I will be building search for specific parameters I.e **searchPersonByName**

**Comment by ibacher** [comment]
@corneliouzbett: Ok. Then let's keep it.

**Comment by corneliouzbett** [comment]
@ibacher I have updated this PR

File: api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImplTest.java [hunk 7]
```diff
@@ -99,4 +102,11 @@ public void findPersonsByNonExistingName_shouldReturnEmptyCollection() {
 		Collection<Person> people = fhirPersonDao.findPersonsByName(NOT_FOUND_NAME);
 		assertThat(people, is(empty()));
 	}
+
+	@Test
+	public void findSimilarPeople_shouldReturnMatchingSimilarPeople() {
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirPersonDaoImplTest.java]: Could you add a tests for what happens when no similar people are found?

**Comment by corneliouzbett** [comment]
@ibacher I have added the tests

**Comment by ibacher** [comment]
Thanks Bett. LGTM.

**Pull request merged**